helpers/setup.sh

```diff
@@ -35,6 +80,21 @@ SERVICE_ACCOUNT_ID="${SERVICE_ACCOUNT_NAME}@${PROJECT_ID}.iam.gserviceaccount.co
 STAGING_DIR="${PWD}"
 KEY_FILE="${STAGING_DIR}/credentials.json"
 
+if [[ ${host_project} != "" ]];
```

Comment:
Logic which is duplicated between `setup.sh` and `cleanup.sh` should be extracted to a shared library file.

Reply:
I'm normally a big proponent for DRY, but for things like simple shell scripts I tend to relax that a bit. Using a shared library reduces the amount of duplicated code but incurs the complexity cost of loading that shared library, which can be pretty obnoxious depending on the bash version and whatnot. We only have two copies of this logic, so I think that we should push back on this refactoring. If we need in the future let's add it, but let's wait for the demand to present itself.

Reply:
After additional review I've seen small issues that affect option parsing in both setup and cleanup; this makes me shift a bit closer towards having a shared file containing common definitions.